src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/Output/AdditionalColumnDataGenerator.cs

```diff
@@ -54,10 +60,15 @@ internal class AdditionalColumnDataGenerator : IAdditionalColumnDataGenerator
             }
             else
             {
-                return new KeyValuePair<string, object>(columnName, property.Value.ToString());
+                return new KeyValuePair<string, object>(columnName, DBNull.Value);
             }
         }
 
+        private static string TruncateOutput(string value, int dataLength) =>
```

Comment:
Please note that there is already a truncate functionality for the Message column in class ```StandardColumnDataGenerator``` which works almost similar (almost because it appends "..." when truncating). 

If we truncate Message and other varchar and nvarchar columns then we should do it in the same way. 

So the suggestion is to refactor the truncate function into a separate class which can be used in both classes which do the truncation.